libraries/botbuilder-adapter-twitter/src/twitter.ts

```diff
@@ -0,0 +1,195 @@
+            id: message.id,
+            timestamp: new Date(),
+            channelId: this.channel,
+            conversation: {
+                id: message.id,
+                isGroup: false,
+                conversationType: null,
+                tenantId: null,
+                name: ''
+            },
+            from: {
+                id: message.user.id,
+                name: message.user.screen_name
+            },
+            recipient: {
+                id: message.in_reply_to_screen_name,
+                name: message.in_reply_to_screen_name
+            },
+            text: message.text,
+            channelData: message,
+            localTimezone: null,
+            callerId: null,
+            serviceUrl: null,
+            listenFor: null,
+            label: message.id,
+            valueType: null,
+            type: null
+        };
+
+        if (activity.type === ActivityTypes.Message) {
+
+            if(message.entities.media != null && message.entities.media.length > 0) {
+                const media = message.entities.media;
+                activity.attachments = [];
+                for (const medium of media) {
+                    let contentType: string;
+                    /*
+                     * Likely need to do a little more parsing of the content/mime type.
+                     */
+                    switch(medium.type) {
+                        case 'photo':
+                            contentType = 'image/png';
+                            break;
+                        case 'video':
+                            contentType = 'video/mpeg';
+                            break;
+                        case 'animated_gif':
+                            contentType = 'image/gif';
+                            break;
+                        default:
+                            throw new Error(`Medium doesn't have an associated content type.`);
+                    }
+                    const attachment = {
+                        contentType: contentType,
+                        contentUrl: medium.media_url_https
+                    };
+                    activity.attachments.push(attachment);
+                }
+            }
+        }
+
+        const context: TurnContext = this.createContext(activity);
+        context.turnState.set('httpStatus', 200);
+        await this.runMiddleware(context, logic);
+        res.status(context.turnState.get('httpStatus'));
+        if (context.turnState.get('httpBody')) {
+            res.send(context.turnState.get('httpBody'));
+        } else {
+            res.end();
+        }
+    }
+
+    protected createContext(request: Partial<Activity>): TurnContext {
+        return new TurnContext(this as any, request);
+    }
+
+    protected createTwitterClient(settings: TwitterSettings): Twitter {
+        return createTwitterClient(settings);
+    }
+
+    protected parseActivity(activity: Partial<Activity>): any {
+
+        const message = { };
```

Comment:
Typecasting? I think it is nicer to init like this.

```
const message: Partial<Message> = {
	status: '',
	in_reply_to_status_id: ''
}
```

Reply:
Fixed.